fenix/app/src/main/java/org/mozilla/fenix/shopping/state/ReviewQualityCheckMiddleware.kt

```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.state
```

Comment:
That's where it started, but then looking at the imports and where it was used (Store, other middlewares), it made sense to be at the center i.e the `state` package. The idea being that `state` package files should not have imports from the `middleware` package but the `middleware` package files can have imports from `state` package. And since middleware is a concept of `Store` so it made sense to be here, wdyt?

Reply:
I have a proposal here where we move the `Store` into its own package similar to `BrowserStore` and then we move this alias to live inside of `ReviewQualityCheckPreferencesMiddleware.kt`.

Reply:
I see that issue that this can be confused as a middleware but the idea was that just that 
`ReviewQualityCheckStore` is a `Store<ReviewQualityCheckState, ReviewQualityCheckAction>`, 
`ReviewQualityCheckMiddleware` is a `Middleware<ReviewQualityCheckState, ReviewQualityCheckAction>`. And since both of these reference the state and action, it's more cohesive that they live together. If we move the store into a `store` package, it's just one file in there and it would have imports to state, action and the middleware. Not a big issue, but trying to keep related things together for more cohesion within the package and less coupling between packages. Happy to get your thoughts on this.

Reply:
Renamed `state` package to `store` 
Moving this along as we discussed it's fine for a store in isolation which won't have multiple action files. Thanks for the discussion 👍